Lib/collections/__init__.py

```diff
@@ -979,6 +979,26 @@ def __contains__(self, key):
 
     # Now, add the methods in dicts but not in MutableMapping
     def __repr__(self): return repr(self.data)
+
+    def __or__(self, other):
+        if isinstance(other, UserDict):
+            return self.__class__(self.data | other.data)
+        if isinstance(other, type(self.data)):
+            return self.__class__(self.data | other)
+        return NotImplemented
+    def __ror__(self, other):
```

Comment:
Please add an empty line between methods to conform PEP 8.

Reply:
(https://www.python.org/dev/peps/pep-0008/#blank-lines)

Reply:
But that would not conform to how the rest of this class is written. I'd leave this be.

Reply:
That was my reasoning. I won't touch this, then.

Reply:
`__ror__` is needed here?

This `__ror__` supports only `UserDict` and `dict`.  But both implements `__or__`.
When is this `__ror__` called?

Reply:
@methane It is used.

The first branch is sometimes taken when the LHS is a `UserDict` subclass that overrides `__or__` and the RHS is a `UserDict`. The second branch is always taken when the LHS is a `dict` and the RHS is a `UserDict`.